lib/core/cursor.js

```diff
@@ -569,13 +628,10 @@ function initializeCursor(cursor, callback) {
         return callback(new MongoError('Topology was destroyed'));
       }
 
-      return cursor.disconnectHandler.addObjectAndMethod(
-        'cursor',
-        cursor,
-        'next',
-        [callback],
-        callback
-      );
+      // TODO: Remove this once we get rid of disconnect handler
+      const err = new MongoError('no connection available');
+      err[mongoErrorContextSymbol].legacyDisconnectHandle = true;
```

Comment:
looking for thoughts on the `legacyDisconnectHandle` item on the `mongoErrorContextSymbol` here. was thinking it might make sense to, if we are using this more freely, make the `mongoErrorContextSymbol` a non-enumerable property of an error so people don't see it all the time and get confused.